### PR TITLE
Refactor events with channels

### PR DESCRIPTION
**Pull request by influx6**
PR updates events channels with changes to stabilize the message formats.

Related issue #51

### CODE REVIEW

File: pushers/channel.go [hunk 1]
```diff
@@ -15,81 +14,58 @@ var log = logging.MustGetLogger("honeytrap:channels")
...
-func NewFilteringChannel(config *config.Config, filters ...Filters) *FilteringChannel {
-	var mc FilteringChannel
-	mc.config = config
-	mc.filters = filters
-
-	return &mc
+// Send delivers the slice of PushMessages and using the internal filters
+// to filter out the desired messages allowed for all registered backends.
+func (mc FilterChannel) Send(msgs ...message.Event) {
+	mc.Backend.Send(mc.Filter.Filter(msgs...)...)
```
Comment (nl5887) [inline pushers/channel.go]: Cannot we change this to single instead of multiple events?

File: pushers/channel.go [hunk 2]
```diff
@@ -15,81 +14,58 @@ var log = logging.MustGetLogger("honeytrap:channels")
...
 
-// FilteringChannel defines a struct which handles the delivery of giving
+// FilterChannel defines a struct which handles the delivery of giving
 // messages to a specific sets of backend channels based on specific criterias.
-type FilteringChannel struct {
-	config   *config.Config
-	backends []Channel
-	filters  []Filters
+type FilterChannel struct {
+	Backend Channel
```
Comment (nl5887) [inline pushers/channel.go]: Can we inherit Backend instead of declaring as var?

File: pushers/channel.go [hunk 3]
```diff
@@ -15,81 +14,58 @@ var log = logging.MustGetLogger("honeytrap:channels")
...
-	// filter messages with all filters.
-	for _, filter := range mc.filters {
-		msgs = filter.Filter(msgs...)
-	}
-
-	// If no message passes our filtering conditions then we have
-	// nothing else to do here.
-	if len(msgs) == 0 {
-		return
+		channels = append(channels, FilterChannel{
```
Comment (nl5887) [inline pushers/channel.go]: What about filterChannel as struct, and FilterChannel method which will return a new FilterChannel{channel, filters}?

File: pushers/backends/slack/slack.go [hunk 4]
```diff
@@ -72,7 +72,7 @@ func init() {
 
 // Send delivers the giving push messages to the required slack channel.
 // TODO: Ask if Send shouldnt return an error to allow proper delivery validation.
-func (mc MessageChannel) Send(messages []message.PushMessage) {
+func (mc MessageChannel) Send(messages ...message.Event) {
```
Comment (influx6) [inline pushers/backends/slack/slack.go]: Yes, we can move the buffering to the channel level and use single message.Events

File: pushers/channel.go (hunk 2 above)
Comment (influx6) [inline pushers/channel.go]: Sure.

**Pull request merged**